### ISSUE TITLE
Bug: [no-redeclare] allow type + namespace

### ISSUE THREAD

**Issue by sviat9440** [typescript-eslint/typescript-eslint#6167]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=C4TwDgpgBAglC8UDOwBOBLAdgcwNwCh9MBDAWwiTGIGNo4BvfKKCADzAHtVgpRIoAQgiiYArqQBGEVAQC++IA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1ukQBNEy8AIY90URNGgdokcGAC+IeUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type A = string;

namespace A {
  export type B = number;
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-redeclare": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No errors with `ignoreDeclarationMerge`

### Actual Result

'A' is already defined. 3:11 - 3:12

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.45.0` |
| `@typescript-eslint/parser` | `5.45.0` |
| `TypeScript` | `4.9.3` |
| `ESLint` | `8.15.0` |
| `node` | `web` |
